lib/Process/Volume/Image/Bundle.pm

```diff
@@ -130,6 +130,15 @@ sub process {
         rename($self->output_filename . ".download", $self->output_filename) || die $!;
     }
 
+    my $check_file_idx = 0;
```

Comment:
This bit is observed behavior on the filesystem, where the previous `rename` is obviously done but the server does not reflect the changes.

Reply:
Interesting. In ingest we shell out to `mv` (https://github.com/hathitrust/feed/blob/master/lib/HTFeed/Storage.pm#L295) and then run an md5 checksum on the files after they've been moved, and I've never noticed any race conditions with that. I don't know that it's worth trying shelling out to `mv` instead of using the `rename` built-in (that has its own issues), but it's at least a data point.